### PR TITLE
Regression test using bare bone vm

### PR DESCRIPTION
**Pull request by persiaAziz**

### CODE REVIEW

File: deploy/vm/test/regression_test/create_ubuntu_vm.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        ssh_public_keys:
+            - path: "/home/{{ vm_user }}/.ssh/authorized_keys"
+              key_data: "{{ public_key_data }}"
+        network_interfaces: "testvm-nic"
+        started: yes
+        state: present
+        image:
+            offer: UbuntuServer
+            publisher: Canonical
+            sku: 16.04-LTS
```
Comment (tniek) [inline deploy/vm/test/regression_test/create_ubuntu_vm.yml]: Didn't we want to do 18.04-LTS?

File: deploy/vm/test/regression_test/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+This directory contains the resources to do regression testing. The current regression test does the following:
+
+1. Creates VM
+2. Installs the required packages and tools for the deployment to work
+
+How to run:
+-----------
+
+ansible-playbook create_ubuntu_vm.yml --extra-vars="{resource_group: "< resource group> ", ssh_private_key_file: "<path to private key file>", ssh_public_key_file: "<path to public key file>", vm_user: "< user name >", allowed_source_ip_prefix: "<range of ip address allowed to ssh to this tester vm. e.g. "X.X.X.X/16">"}"
```
[the remark this answers is not in the document]
Comment (persiaAziz) [reply deploy/vm/test/regression_test/README.md]: you will run this in the console. all you need is the quotes around the name of the variables you are passing, like this

```
ansible-playbook create_ubuntu_vm.yml --extra-vars="{resource_group: "persia-test2", ssh_private_key_file: "/tmp/sshkey", ssh_public_key_file: "/tmp/sshkey.pub", vm_user: "persia"}"
```

File: deploy/vm/test/regression_test/create_ubuntu_vm.yml (hunk 1 above)
Comment (persiaAziz) [reply deploy/vm/test/regression_test/create_ubuntu_vm.yml]: We did, When I did searched the list of images available and their sku, I just found 16.04, but I see 18.04 in the portal. I wonder why azure cli does not show me that. I will see if just editing the sku to be 18.04-LTS works

 {
    "offer": "SLES",
    "publisher": "SUSE",
    "sku": "12-SP2",
    "urn": "SUSE:SLES:12-SP2:latest",
    "urnAlias": "SLES",
    "version": "latest"
  },
  {
    "offer": "UbuntuServer",
    "publisher": "Canonical",
    "sku": "16.04-LTS",
    "urn": "Canonical:UbuntuServer:16.04-LTS:latest",
    "urnAlias": "UbuntuLTS",
    "version": "latest"
  },
  {
    "offer": "WindowsServer",
    "publisher": "MicrosoftWindowsServer",
    "sku": "2016-Datacenter",
    "urn": "MicrosoftWindowsServer:WindowsServer:2016-Datacenter:latest",
    "urnAlias": "Win2016Datacenter",
    "version": "latest"
  },

Comment (tniek) [reply deploy/vm/test/regression_test/create_ubuntu_vm.yml]: "imageReference": {
    "publisher": "Canonical",
    "offer": "UbuntuServer",
    "sku": "18.04-LTS",
    "version": "latest"
}

Comment (persiaAziz) [reply deploy/vm/test/regression_test/create_ubuntu_vm.yml]: What command did you run to get this list?

Comment (tniek) [reply deploy/vm/test/regression_test/create_ubuntu_vm.yml]: Before you deploy the VM in portal, click on "Download a template for automation" - it's in the JSON.
https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-quickstart-create-templates-use-the-portal

**Pull request merged**